build/ci/create_demo_pr.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash -xe
+set +o pipefail
+
+sed -i "s+$current_operator_image+$wanted_operator_image+g" $csv_file
```

Comment:
considered using a structure-based tool, e.g. `yq`

Reply:
did you use [`|=`](https://mikefarah.gitbook.io/yq/operators/assign-update#updated-multiple-paths)?
what is the error?

Reply:
which error?

Reply:
I assumed it didn't succeed because of an error.
could you write here what `yq` command should, but doesn't work?

Reply:
I don't remember the command that I used (it was two weeks ago) but I remember that the `yq` command didn't parse our YAML good in the alm-examples field

Reply:
oh, then alm-examples value is a json, so it probably needs an additional `jq` command.
let me know if you need any help with that

Reply:
done
but I didn't managed to do it in a function because I couldn't pass variables to the field section

Reply:
please share the code that doesn't work (with error, if exists)

Reply:
this was my last try
`yq --arg fieldtest $field -y '.[$fieldtest] |= "ni"' ibm-block-csi-operator.v1.6.0.clusterserviceversion.yaml > matan.yaml`

Reply:
for latest `yq` version, this works for me (with `export` before, and without a `temp.yaml`):
`yq eval ".spec.relatedImages[0].image |= env(operator_image_for_test)" ibm-block-csi-operator.v1.6.0.clusterserviceversion.yaml -i`

Reply:
done
but still didn't manage to fil the field section with parameter

Reply:
```suggestion
declare -a operator_image_fields=(".spec.install.spec.deployments[0].spec.template.spec.containers[0].image" ".metadata.annotations.containerImage" ".spec.relatedImages[0].image")

for image_field in "${operator_image_fields[@]}"
do
    yq eval "$image_field |= env(operator_image_for_test)" $(basename $csv_file) -i
done
```